test/services/ReviewServiceSpec.scala

```diff
@@ -98,6 +103,28 @@ class ReviewServiceSpec extends BaseSpec with MockFactory with ReviewData with A
       }
     }
 
+    "there are entries to return but another published process has the same process code" should {
+      "return a duplicate key error" in new Test {
+
+        MockPublishedService
+          .getByProcessCode(approvalProcess.meta.processCode)
+          .returns(Future.successful(Right(PublishedProcess("anotherId", 1, ZonedDateTime.now(), Json.obj(), "", approvalProcess.meta.processCode))))
+
+        MockApprovalRepository
+          .getById(validId)
+          .returns(Future.successful(Right(approvalProcess)))
+
+        MockApprovalProcessReviewRepository
```

Comment:
I don't think the method getByIdVersionAndType is actually invoked.

Reply:
I removed that line and test still passes.  Will push up with other changes when you have finished review